core-tests/shared/src/test/scala/zio/FiberSpec.scala

```diff
@@ -119,7 +119,29 @@ object FiberSpec extends ZIOBaseSpec {
       testM("collectAll") {
         assertM(Fiber.collectAll(fibers).join)(anything)
       }
-    ) @@ sequential
+    ) @@ sequential,
+    suite("track blockingOn")(
+      testM("in await") {
+        for {
+          p    <- Promise.make[Nothing, Unit]
+          f1   <- p.await.fork
+          f1id <- f1.id
+          f2   <- f1.await.fork
+          dump <- f2.dump
+        } yield assert(dump.status)(
+          isSubtype[Fiber.Status.Suspended](hasField("blockingOn", _.blockingOn, equalTo(List(f1id))))
+        )
+      } @@ eventually,
```

Comment:
I would generally use `flaky` instead of `eventually`.

Can you use latches to remove the flakiness from this one and the below one?

Reply:
I have added latches. Strictly speaking there could still be a race condition, but the probability is low and I was not able to cause one.

Reply:
... until it ran on CI. 🤣

Reply:
Using latches is difficult here: As I test blocking behavior, I must release the latch right before starting to block, so there could still be a race condition. One way of making the behavior deterministic is using a single threaded `Executor`. That makes the test platform specific, which I do not like either. I've resorted to retrying while the `Fiber.Status` still is `Running`. What do you think?

Reply:
If the test must be flaky, then I would probably use `flaky` to indicate this.

Reply:
👍